app/controllers/concerns/sufia/users_controller_behavior.rb

```diff
@@ -68,17 +68,17 @@ def update_directory?
   end
 
   def toggle_trophy
-    unless current_user.can? :edit, params[:parent_id]
+    work_id = params[:work_id]
+    unless current_user.can? :edit, GenericWork.find(work_id)
```

Comment:
I think you can omit `GenericWork.find`, just `can? :edit, work_id` ought to do the right thing and will be faster.